packages/protocol/scripts/foundry/constants.sh

```diff
@@ -0,0 +1,61 @@
+export FROM_ACCOUNT="0x$FROM_ACCOUNT_NO_ZERO" # Anvil default account (1)
+export FROM_ACCOUNT_PRIVATE_KEY_NO_ZERO="ac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80" # Anvil default account (1)
+export FROM_ACCOUNT_PRIVATE_KEY="0x$FROM_ACCOUNT_PRIVATE_KEY_NO_ZERO" # Anvil default account (1)
+
+# Anvil configurations (Source: https://book.getfoundry.sh/reference/anvil/)
+export ANVIL_PORT=8546
+export ANVIL_RPC_URL="http://127.0.0.1:$ANVIL_PORT"
+export GAS_LIMIT=50000000
+export CODE_SIZE_LIMIT=245760 # EIP-170: Contract code size limit in bytes. Useful to increase for tests. [default: 0x6000 (~25kb)]
+export BALANCE=60000 # Set the balance of the accounts. [default: 10000]
+export STATE_INTERVAL=1 # Interval in seconds at which the state and block environment is to be dumped to disk.
+export STEPS_TRACING="--steps-tracing" #  Steps tracing used for debug calls returning geth-style traces. Enable: "--steps-tracing" / Disable: ""
+
+# Forge migration script configurations (Source: https://book.getfoundry.sh/reference/forge/forge-script)
+export MIGRATION_SCRIPT_PATH="migrations_sol/Migration.s.sol" # Path to migration script
+export TARGET_CONTRACT="Migration" #  The name of the contract you want to run.
+export BROADCAST="--broadcast" # Broadcasts the transactions. Enable: "--broadcast" / Disable: ""
+export SKIP_SIMULATION="" # Skips on-chain simulation. Enable: "--skip-simulation" / Disable: ""
+export NON_INTERACTIVE="--non-interactive" # Remove interactive prompts which appear if the contract is near the EIP-170 size limit.
+export VERBOSITY_LEVEL="-vvv" # Pass multiple times to increase the verbosity (e.g. -v, -vv, -vvv).
+export REGISTRY_OWNER_ADDRESS=$FROM_ACCOUNT_NO_ZERO
+
+# Foundry directories and file names
+export L1_DEVCHAIN_FILE_NAME="devchain.json" # Name of the file that will be published to NPM
+export L2_DEVCHAIN_FILE_NAME="l2-devchain.json" # Name of the file that will be published to NPM
+export TMP_FOLDER="$PWD/.tmp"
+export TEMP_DIR="$PWD/.tmp/libraries"
+export ANVIL_FOLDER="$TMP_FOLDER/devchain"
+
+# Contract addresses
+export REGISTRY_ADDRESS="0x000000000000000000000000000000000000ce10"
+export PROXY_ADMIN_ADDRESS='0x4200000000000000000000000000000000000018' # This address is defined in `IsL2Check.sol`
+
+# Contract configurations
+export COMMUNITY_REWARD_FRACTION="100000000000000000000" # 0.01 in fixidity format
+export CARBON_OFFSETTING_PARTNER="0x22579CA45eE22E2E16dDF72D955D6cf4c767B0eF"
+export CARBON_OFFSETTING_FRACTION="10000000000000000000" # 0.001 in fixidity format
+export REGISTRY_STORAGE_LOCATION="0xb53127684a568b3173ae13b9f8a6016e243e63b6e8ee1178d6a717850b5d6103" # Position is bytes32(uint256(keccak256("eip1967.proxy.admin")) - 1);
+export CELO_DISTRIBUTION_SCHEDULE_INITIAL_BALANCE="10000" # Arbitrary balance choosen at random. During the real L2 genesis, the VM will calculate and set an appropriate balance.
```

Comment:
TODO: Check what balance I should manually set for the `CeloDistributionSchedule` during activation.

Reply:
GoldToken.CELO_SUPPLY_CAP - GoldToken.totalSupply()

But `totalSupply()` on devchain might be low, so might be worth fetching `totalSupply()` on Alfajores or Mainnet.

Or set an arbitrary number: 
- 600M was allocated at genesis (that's at least in `totalSupply()` so anything above 600M)
- 200M is supposed to be allocated in the 15 years since genesis (so 800M is ceilling)
Anything between 600M and 800M, like 700M.

Reply:
`GoldToken.totalSupply()` on the L1 devchain is too low (`0`): 

Start L1 devchain (from NPM artifact):

```sh
anvil \
--port 8546 \
--state ~/Downloads/devchain-11107-2024-07-08/devchain.json
```

In a different terminal:

```sh
# Get address of GoldToken
$ cast call \
0x000000000000000000000000000000000000ce10 \
"getAddressForStringOrDie(string calldata identifier)(address)" \
"GoldToken" \
--rpc-url=http://127.0.0.1:8546
0x2365F470a061B8333BAdE225eb7B34E2f9A3B4B2

# Call `totalSupply()` on `GoldToken.sol`
$ cast call \
0x2365F470a061B8333BAdE225eb7B34E2f9A3B4B2 \
"totalSupply()(uint256)" \
--rpc-url=http://127.0.0.1:8546
0
```

Reply:
`GoldToken.totalSupply()` on the L1 Mainnet is 695,313,643 CELO:

```sh
# Get address of GoldToken
$ cast call \
0x000000000000000000000000000000000000ce10 \
"getAddressForStringOrDie(string calldata identifier)(address)" \
"GoldToken" \
--rpc-url https://forno.celo.org
0x471EcE3750Da237f93B8E339c536989b8978a438

# Call `totalSupply()` on `GoldToken.sol`
$ cast call \
0x471EcE3750Da237f93B8E339c536989b8978a438 \
"totalSupply()(uint256)" \
--rpc-url https://forno.celo.org
695313643195035937058427065 [6.953e26]

# Convert units from wei to ether
$ cast to-unit 695313643195035937058427065 ether
695313643.195035937058427065
```

👉 **I'll manually set the amount at 700M CELO.**

For context, `GoldToken.totalSupply()` on the L1 Alfajores is not representative (109,959,073 CELO):

```sh
$ cast call \
0x000000000000000000000000000000000000ce10 \
"getAddressForStringOrDie(string calldata identifier)(address)" \
"GoldToken" \
--rpc-url https://alfajores-forno.celo-testnet.org
0xF194afDf50B03e69Bd7D057c1Aa9e10c9954E4C9

$ cast call \
0xF194afDf50B03e69Bd7D057c1Aa9e10c9954E4C9 \
"totalSupply()(uint256)" \
--rpc-url https://alfajores-forno.celo-testnet.org
109959073349267550845576790 [1.099e26]

$ cast to-unit 109959073349267550845576790 ether
109959073.349267550845576790
```

Reply:
That's done in https://github.com/celo-org/celo-monorepo/pull/11107/commits/06e7fe747dcde8f23f05521d7cb5ae18fdc6e35b

Reply:
I mistakenly set the initial balance of the CeloDistributionSchedule to `GoldToken.totalSupply()`, instead of `GoldToken.CELO_SUPPLY_CAP - GoldToken.totalSupply()`. This is updated and fixed here: 
- https://github.com/celo-org/celo-monorepo/pull/11140